racket/collects/net/http-client.rkt

```diff
@@ -97,10 +103,15 @@
   ;; (thread (λ () (copy-port log-i to (current-error-port))))
 
   (set-http-conn-to! hc to)
-  (set-http-conn-from! hc from))
+  (set-http-conn-from! hc from)
+
+  (set-http-conn-auto-reconnect?! hc auto-reconnect?)
+  (set-http-conn-auto-reconnect-host! hc host-bs)
```

Comment:
I believe you’re missing this: https://github.com/racket/racket/blob/6632beeca98de0dda9b56121ee7ebed517a3a1ba/racket/collects/net/http-client.rkt#L104

The function `http-conn-close!` overwrites the `host` and then it’s lost—we don’t have access to it when trying to reconnect.